### ISSUE TITLE
Spinnaker 1.19.4 - service accounts created in Front50 not synced to Fiat

### ISSUE THREAD

**Issue by sofam** [spinnaker/spinnaker#5691]
### Issue Summary:
Service accounts are not synced to Fiat

### Cloud Provider(s):
Google Cloud Platform

### Environment:
Spinnaker 1.19.4 running on GKE

### Feature Area:
Service accounts, Fiat, Front50

### Description:
When creating a new service account, it is not synced over to Fiat.

### Steps to Reproduce:
````
/# curl -X POST \
>   -H "Content-type: application/json" \
>   -d '{ "name": "vml2@e-conomic.net", "memberOf": ["vml"] }' \
> http://spin-front50.spinnaker:8080/serviceAccounts
{"name":"vml2@e-conomic.net","lastModified":1587138472000,"lastModifiedBy":"anonymous","memberOf":["vml"]}
```
After syncing the roles, this shows up as a 404 in fiat. We noticed this when trying to switch to managed service accounts. And this is a stable reproduce of the issue.

```
root@nginx-8586cf59-5tn8z:/# curl -XGET http://spin-fiat.spinnaker:7003/authorize/vml2@e-conomic.net
{
  "timestamp" : 1587138814886,
  "status" : 404,
  "error" : "Not Found",
  "message" : "No message available"
}
### Additional Details:

**Comment by ajordens** [comment]
Likely a duplicate of #5687

**Comment by sofam** [comment]
I tried reproducing this with a local install of Spinnaker with github authn and authz, using openjdk 8, it's not possible to reproduce in JDK 8.

I haven't gotten the local install to run with openjdk 11 locally. We will try to deploy spinnaker in production with image-variant java8


### Steps I tried to reproduce with jdk8

Adding service account
```
➜  logs  curl -X POST \                                                                     
  -H "Content-type: application/json" \
  -d '{ "name": "sekret-service-account2@spinnaker-test.net", "memberOf": ["test"] }' http://localhost:8080/serviceAccounts
{"name":"sekret-service-account2@spinnaker-test.net","lastModified":1587168323000,"lastModifiedBy":"anonymous","memberOf":["test"]}%                                                                        ➜  logs  curl -X POST http://localhost:7003/roles/sync                                        
```


Syncing                                                                                                                                                                                                    
```
➜  logs  curl -X POST http://localhost:7003/roles/sync
3% 
```                                                                                                                                                                                                         

Lastly checking the authorizations for the account
```
➜  logs  curl -X GET http://localhost:7003/authorize/sekret-service-account2@spinnaker-test.net
{
  "name" : "sekret-service-account2@spinnaker-test.net",
  "accounts" : [ {
    "name" : "minikube",
    "authorizations" : [ "WRITE", "READ" ]
  } ],
  "applications" : [ {
    "name" : "ratings",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "productpage",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "kubernetes",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "zipkin",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "details",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "coredns",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "istiod",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "prometheus",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "jaeger",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "tracing",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "kube",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "kiali",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "grafana",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "istio",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  }, {
    "name" : "reviews",
    "authorizations" : [ "READ", "WRITE", "EXECUTE", "CREATE" ]
  } ],
  "serviceAccounts" : [ {
    "name" : "sekret-service-account@spinnaker-test.net",
    "memberOf" : [ "test" ]
  }, {
    "name" : "sekret-service-account2@spinnaker-test.net",
    "memberOf" : [ "test" ]
  } ],
  "roles" : [ {
    "name" : "test",
    "source" : "EXTERNAL"
  } ],
  "buildServices" : [ ],
  "admin" : false,
  "legacyFallback" : false,
  "allowAccessToUnknownApplications" : false
}%                                                         
```

Works 1.19.4 with Java 8

**Comment by sofam** [comment]
### Java 11 reproduce which works
On a local install:

```
➜  scripts (⎈ minikube:default) curl -X POST \                                                                     
  -H "Content-type: application/json" \
  -d '{ "name": "sekret-service-account3@spinnaker-test.net", "memberOf": ["test"] }' http://localhost:8080/serviceAccounts
curl: (6) Could not resolve host:  
zsh: command not found: -H
➜  scripts (⎈ minikube:default) curl -X POST \-H "Content-type: application/json" \       
  -d '{ "name": "sekret-service-account3@spinnaker-test.net", "memberOf": ["test"] }' http://localhost:8080/serviceAccounts
{"name":"sekret-service-account3@spinnaker-test.net","lastModified":1587201139000,"lastModifiedBy":"anonymous","memberOf":["test"]}%                                                                        ➜  scripts (⎈ minikube:default) curl -X POST http://localhost:7003/roles/sync      
4%                                                                                                                                                                                                          ➜  scripts (⎈ minikube:default) curl -X GET http://localhost:7003/authorize/sekret-service-account3@spinnaker-test.net
{
  "timestamp" : 1587201156045,
  "status" : 404,
  "error" : "Not Found",
  "message" : "No message available"
}%                                      
```

**Comment by ajordens** [comment]
Going to close this in favor of #5687, should be fixed in 1.19.6 (against java11 and java8).

Thanks @danopia for for investigation and fix!
